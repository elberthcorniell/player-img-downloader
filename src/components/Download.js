import { handleFileDownload } from '../utils';
import { useState } from 'react';
import { Button } from 'react-bootstrap';
import PropTypes from 'prop-types';

const Download = ({ style, url, children }) => {
    const [loading, setLoading] = useState()
    return < Button
        variant='success'
        style={style}
        onClick={async () => {
            setLoading(true);
            await handleFileDownload(url);
            setLoading(false);
        }}
    >
        {loading ? <img src="https://inverte.io/images/spinner.gif" alt="loader" height="30px" /> : children}
    </Button >
}

Download.propTypes = {
    style: PropTypes.objectOf(PropTypes.oneOfType([
        PropTypes.string,
        PropTypes.number
    ])),
    url: PropTypes.string,
    children: PropTypes.oneOfType([
        PropTypes.string,
        PropTypes.element
    ]),
}

Download.defaultProps = {
    style: {},
    url: '',
    children: <></>,
}

export default Download;
