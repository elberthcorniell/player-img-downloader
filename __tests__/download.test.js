import { render, screen } from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';
import Download from '../src/components/Download';

describe('Download Button', () => {
    test('Render as expected', () => {
        render(<Download />)
        expect(document.body.innerHTML).toMatchSnapshot();
    });

    test('Render as expected', () => {
        render(<Download>test</Download>)
        const textElement = screen.getAllByText('test');
        expect(typeof textElement[0]).not.toBe('undefined');
    });
});