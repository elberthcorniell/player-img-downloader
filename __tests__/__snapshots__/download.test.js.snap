// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Download Button Render as expected 1`] = `"<div><button type=\\"button\\" class=\\"btn btn-success\\"></button></div>"`;
